pkg/testing/fixture.go

```diff
@@ -714,24 +714,100 @@ func (e *ExecErr) Unwrap() error {
-	out, err := f.Exec(ctx, []string{"status", "--output", "json"}, opts...)
-	status := AgentStatusOutput{}
-	if uerr := json.Unmarshal(out, &status); uerr != nil {
-		return AgentStatusOutput{},
-			fmt.Errorf("could not unmarshal agent status output: %w:\n%s", errors.Join(uerr, err), out)
-	} else if status.IsZero() {
-		return status, fmt.Errorf("agent status output is empty: %w", err)
+type statusOpts struct {
+	noRetry       bool
+	retryTimeout  time.Duration
+	retryInterval time.Duration
+
+	cmdOptions []process.CmdOption
+}
+
+type statusOpt func(*statusOpts)
+
+// WithNoRetry disables the retry logic in ExecStatus function call.
+func WithNoRetry() func(opt *statusOpts) {
+	return func(opt *statusOpts) {
+		opt.noRetry = true
 	}
+}
 
-	return status, nil
+// WithRetryTimeout adjusts the retry timeout from the default value of one minute.
+func WithRetryTimeout(duration time.Duration) func(opt *statusOpts) {
+	return func(opt *statusOpts) {
+		opt.retryTimeout = duration
+	}
+}
+
+// WithRetryInterval adjusts the retry interval from the default value of one second.
+func WithRetryInterval(duration time.Duration) func(opt *statusOpts) {
+	return func(opt *statusOpts) {
+		opt.retryInterval = duration
+	}
+}
+
+// WithCmdOptions adjusts the options of the command when status is called.
+func WithCmdOptions(cmdOptions ...process.CmdOption) func(opt *statusOpts) {
+	return func(opt *statusOpts) {
+		opt.cmdOptions = append(opt.cmdOptions, cmdOptions...)
+	}
+}
+
+// ExecStatus executes `elastic-agent status --output=json`.
+//
+// Returns the parsed output and the error from the execution. Keep in mind the agent exits with status 1 if it's
+// unhealthy, but it still outputs the status successfully. This call does require that the Elastic Agent is running
+// and communication over the control protocol is working.
+//
+// By default, retry logic is applied. Use WithNoRetry to disable this behavior. WithRetryTimeout and WithRetryInterval
+// can be used to adjust the retry logic timing. The default retry timeout is one minute and the default retry
+// interval is one second.
+//
+// An empty AgentStatusOutput and non nil error means the output could not be parsed. As long as we get some output,
+// we don't return any error. It should work with any 8.6+ agent
+func (f *Fixture) ExecStatus(ctx context.Context, opts ...statusOpt) (AgentStatusOutput, error) {
+	var opt statusOpts
+	opt.retryTimeout = 1 * time.Minute
+	opt.retryInterval = 1 * time.Second
+	for _, o := range opts {
+		o(&opt)
+	}
+
+	if opt.noRetry {
+		out, err := f.Exec(ctx, []string{"status", "--output", "json"}, opt.cmdOptions...)
+		status := AgentStatusOutput{}
+		if uerr := json.Unmarshal(out, &status); uerr != nil {
+			return AgentStatusOutput{},
+				fmt.Errorf("could not unmarshal agent status output: %w:\n%s", errors.Join(uerr, err), out)
+		} else if status.IsZero() {
+			return status, fmt.Errorf("agent status output is empty: %w", err)
+		}
```

Comment:
I believe that a `return status, nil` is missing from here

Reply:
No this should be an error. It should not be zero, and we want it to error. This was not changed from the original code.

https://github.com/elastic/elastic-agent/blob/ca055614fb89faf244fe67d3d19d5302884cbea3/pkg/testing/fixture.go#L731

Reply:
ok let me be more representantive

```go
if opt.noRetry {
		out, err := f.Exec(ctx, []string{"status", "--output", "json"}, opt.cmdOptions...)
		status := AgentStatusOutput{}
		if uerr := json.Unmarshal(out, &status); uerr != nil {
			return AgentStatusOutput{},
				fmt.Errorf("could not unmarshal agent status output: %w:\n%s", errors.Join(uerr, err), out)
		} else if status.IsZero() {
			return status, fmt.Errorf("agent status output is empty: %w", err)
		}
                // <--- here a return is missing I believe (happy path no error)
	}
```

Reply:
I see what you mean, sorry that was not clear from the comment. I updated the code to your suggestion to remove the code duplication, which had this in there. Should be good now.